provider/mem/mem.go

```diff
@@ -201,6 +201,12 @@ func (a *Alerts) Put(alerts ...*types.Alert) error {
 
 		a.mtx.Lock()
 		for _, l := range a.listeners {
+			select {
+			case <-l.done:
+				continue
+			default:
+			}
+
```

Comment:
Would it be better to rely only on alerts channel being open? e.g.

```suggestion
		for _, l := range a.listeners {
			if a.alerts == nil {
			  continue
			}

			select {
			case l.alerts <- alert:
			}
```
and elsewhere when the channel is closed, 
```
select {
case <-l.done:
  close(l.alerts)
  l.alerts = nil
  // perhaps even  l.done = nil
}
```

Reply:
Because of the lock protection, the problem described by the pr will not occur, so I  closed this pr. Thanks.